reinforcement_learning/trainer/common/actor_critic.py

```diff
@@ -0,0 +1,211 @@
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import tensorflow as tf
+
+
+class ActorNetwork(object):
+    """
+    Input to the network is the state, output is the action
+    under a deterministic policy.
+    """
+
+    def __init__(self,
+                 sess,
```

Comment:
it is a little awkward to have a network/model object holding a session.

Reply:
Yeah, I felt the same.
The other option is to pass the session as an argument to every method in agent.py.
Which one do you think is better?
Or do we have other design options?

Reply:
it is probably more natural from the standpoint of TF to have the agent expose output tensors, and have the session in the training loop run those tensors.